### PR TITLE
Swich to cssnano and generate minified css+sourcemaps

### LINKED ISSUE #361: gulp-minify-css is deprecated, please use gulp-cssnano instead
### Steps to reproduce:

``` sh
$ git clone git@github.com:mozilla/idea-town.git
$ cd idea-town
$ npm i
```
### Actual results:

``` sh
$ rm -rf node_modules && npm i

npm WARN deprecated gulp-minify-css@1.2.3: Please use gulp-cssnano instead.
```

### LINKED ISSUE #362: Trying to generate a production quality stylesheet implodes and throws errors
### Steps to reproduce:
1. Clone the **idea-town** repo.
2. Change the `IS_DEBUG` const on line 27 of the gulpfile.js to `false`.
3. Run `$ gulp styles` from The Terminal.
### Actual results:

``` sh
$ gulp styles
[15:37:40] Using gulpfile ~/dev/tmp/del/idea-town/gulpfile.js
[15:37:40] Starting 'styles'...
/Users/pdehaan/dev/tmp/del/idea-town/node_modules/gulp-minify-css/node_modules/vinyl-sourcemaps-apply/node_modules/source-map/lib/source-map-consumer.js:703
        throw new Error('"' + aSource + '" is not in the SourceMap.');
        ^

Error: "node_modules/node-normalize-scss/_normalize.scss" is not in the SourceMap.
    at SourceMapConsumer_sourceContentFor [as sourceContentFor] (/Users/pdehaan/dev/tmp/del/idea-town/node_modules/gulp-minify-css/node_modules/vinyl-sourcemaps-apply/node_modules/source-map/lib/source-map-consumer.js:703:15)
    at SourceMapGenerator.<anonymous> (/Users/pdehaan/dev/tmp/del/idea-town/node_modules/gulp-minify-css/node_modules/vinyl-sourcemaps-apply/node_modules/source-map/lib/source-map-generator.js:229:42)
    at Array.forEach (native)
    at SourceMapGenerator_applySourceMap [as applySourceMap] (/Users/pdehaan/dev/tmp/del/idea-town/node_modules/gulp-minify-css/node_modules/vinyl-sourcemaps-apply/node_modules/source-map/lib/source-map-generator.js:228:34)
    at applySourceMap (/Users/pdehaan/dev/tmp/del/idea-town/node_modules/gulp-minify-css/node_modules/vinyl-sourcemaps-apply/index.js:27:15)
    at /Users/pdehaan/dev/tmp/del/idea-town/node_modules/gulp-minify-css/index.js:56:13
    at whenSourceMapReady (/Users/pdehaan/dev/tmp/del/idea-town/node_modules/gulp-minify-css/node_modules/clean-css/lib/clean.js:139:16)
    at /Users/pdehaan/dev/tmp/del/idea-town/node_modules/gulp-minify-css/node_modules/clean-css/lib/clean.js:151:18
    at fromString (/Users/pdehaan/dev/tmp/del/idea-town/node_modules/gulp-minify-css/node_modules/clean-css/lib/utils/input-source-map-tracker.js:33:10)
    at InputSourceMapStore.track (/Users/pdehaan/dev/tmp/del/idea-town/node_modules/gulp-minify-css/node_modules/clean-css/lib/utils/input-source-map-tracker.js:236:5)
    at Object.whenDone (/Users/pdehaan/dev/tmp/del/idea-town/node_modules/gulp-minify-css/node_modules/clean-css/lib/clean.js:145:44)
    at processNext (/Users/pdehaan/dev/tmp/del/idea-town/node_modules/gulp-minify-css/node_modules/clean-css/lib/imports/inliner.js:105:13)
    at importFrom (/Users/pdehaan/dev/tmp/del/idea-town/node_modules/gulp-minify-css/node_modules/clean-css/lib/imports/inliner.js:79:10)
    at ImportInliner.process (/Users/pdehaan/dev/tmp/del/idea-town/node_modules/gulp-minify-css/node_modules/clean-css/lib/imports/inliner.js:38:10)
    at /Users/pdehaan/dev/tmp/del/idea-town/node_modules/gulp-minify-css/node_modules/clean-css/lib/clean.js:120:41
    at doNTCallback0 (node.js:419:9)
```

### PR DESCRIPTION
**Pull request by pdehaan**
Fixes #361 
Fixes #362

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -110,11 +110,10 @@ gulp.task('styles', ['sass-lint'], function stylesTask() {
       ]
     }).on('error', sass.logError))
     .pipe(autoprefixer('last 2 versions'))
-    .pipe(sourcemaps.write({sourceRoot: SRC_PATH + 'styles'}))
-    .pipe(gulp.dest(DEST_PATH + 'styles'))
       // don't minify in development
       .pipe(gulpif(!IS_DEBUG, rename({ suffix: '.min' })))
       .pipe(gulpif(!IS_DEBUG, minifycss()))
+      .pipe(gulpif(!IS_DEBUG, sourcemaps.write('.')))
```
Comment (pdehaan) [inline gulpfile.js:116]: I'm only writing the CSS sourcemap here if we're generating minified CSS.

File: idea_town/frontend/static-src/styles/_state.scss [hunk 2]
```diff
@@ -84,6 +84,7 @@
   from {
     background-position: 0 0;
   }
+
```
Comment (pdehaan) [inline idea_town/frontend/static-src/styles/_state.scss:87]: opportunistic fix of a sass-lint warning:

```
idea_town/frontend/static-src/styles/_state.scss
  87:1  warning  Space expected between blocks  empty-line-between-blocks
```

File: idea_town/frontend/static-src/styles/modules/_modals.scss [hunk 3]
```diff
@@ -85,7 +85,7 @@
   background-color: $white;
   border-radius: $feedback-modal-border-radius;
   left: 50%;
-  margin-left: -($feedback-modal-width / 2);
+  margin-left: ($feedback-modal-width / - 2);
```
Comment (pdehaan) [inline idea_town/frontend/static-src/styles/modules/_modals.scss:88]: opportunistic fix of sass-lint warning:

```
idea_town/frontend/static-src/styles/modules/_modals.scss
  88:16  warning  Space expected around operator  space-around-operator
```

Gross fix, but was only way I could suppress the warning. ¯\_(ツ)_/¯

**Comment by meandavejustice** [comment]
Looks great, thanks @pdehaan

**Pull request merged**